src/software/geom/util.cpp

```diff
@@ -440,23 +442,24 @@ std::vector<std::pair<Point, Angle>> angleSweepCirclesAll(
         Vector inter      = lineIntersection(src, src + ray, p1, p2).value();
 
         // Offset the final values by the start angle
-        result.emplace_back(std::make_pair(inter, range_end - range_start));
+        result.emplace_back(Shot(inter, range_end - range_start));
     }
 
     return result;
 }
 
-std::optional<std::pair<Point, Angle>> angleSweepCircles(
-    const Vector &src, const Vector &p1, const Vector &p2,
-    const std::vector<Vector> &obstacles, const double &radius)
+std::optional<Shot> angleSweepCircles(const Vector &src, const Vector &p1,
+                                      const Vector &p2,
+                                      const std::vector<Vector> &obstacles,
+                                      const double &radius)
 {
     // Get all possible shots we could take
-    std::vector<std::pair<Point, Angle>> possible_shots =
+    std::vector<Shot> possible_shots =
         angleSweepCirclesAll(src, p1, p2, obstacles, radius);
 
     // Sort by the interval angle (ie. the open angle the shot is going through)
     std::sort(possible_shots.begin(), possible_shots.end(),
-              [](auto p1, auto p2) { return p1.second > p2.second; });
+              [](auto s1, auto s2) { return s1.angle > s2.angle; });
```

Comment:
The code outside the `shot` class should not have direct access to the member variables. After changes, the code would be something similar to `s1.getAngle() >s2.getAngle()`

Reply:
Updated